lib/reference/properties.js

```diff
@@ -311,55 +320,70 @@ const longhandSubPropertiesOfShorthandProperties = new Map([
 			'mask-origin',
 			'mask-clip',
 			'mask-composite',
-		]),
+		],
 	],
 	[
 		'outline',
-		new Set([
+		[
 			// prettier-ignore
 			'outline-color',
 			'outline-style',
 			'outline-width',
-		]),
+		],
 	],
 	[
 		'padding',
-		new Set([
+		[
 			// prettier-ignore
 			'padding-top',
+			'padding-right',
 			'padding-bottom',
 			'padding-left',
-			'padding-right',
-		]),
+		],
 	],
 	[
 		'text-decoration',
-		new Set([
+		[
 			// prettier-ignore
 			'text-decoration-color',
 			'text-decoration-style',
 			'text-decoration-line',
-		]),
+		],
 	],
 	[
 		'text-emphasis',
-		new Set([
+		[
 			// prettier-ignore
 			'text-emphasis-style',
 			'text-emphasis-color',
-		]),
+		],
 	],
 	[
 		'transition',
-		new Set([
-			'transition-delay',
-			'transition-duration',
+		[
+			// prettier-ignore
 			'transition-property',
+			'transition-duration',
 			'transition-timing-function',
-		]),
+			'transition-delay',
+		],
 	],
 ]);
 
+function convertToLonghandSubPropertiesOfShorthandProperties() {
+	const map = new Map();
+
+	for (const [shorthand, orderedLonghand] of orderedLonghandSubPropertiesOfShorthandProperties) {
+		map.set(shorthand, new Set(orderedLonghand));
+	}
+
+	return map;
+}
+
+/** @type {import('stylelint').LonghandSubPropertiesOfShorthandProperties} */
+const longhandSubPropertiesOfShorthandProperties =
+	convertToLonghandSubPropertiesOfShorthandProperties();
```

Comment:
I did this instead of using `.entries().map(...)` and then making a `Map` because it seems like `Object.entries` didn't preserve the type of `orderedLonghandSubPropertiesOfShorthandProperties` properly.

This feels a bit hacky - is there better code I could be doing here (or something I've made a mistake  on)